docs/features/asc1/teal/index.md

```diff
@@ -13,17 +13,17 @@ Some of the opcodes in TEAL are only valid for a specific contract type. These a
+Currently, Algorand supports versions 1 - 4 of TEAL. When writing TEAL Version 2 or higher contracts make sure to add `#pragma version #` where # should be replaced by the specific number, as the first line of the program. If this line does not exists, the protocol will treat the contract as a version 1 contract. If upgrading a contract to version 2 or higher it is important to verify you are checking the `RekeyTo` property of all transactions that are attached to the contract. See the [TEAL guidelines](../../../reference/teal/guidelines.md) for more suggestions to help prevent a contract from being compromised. 
 
 # Getting Transaction Properties
-The primary purpose of a TEAL program is to return either true or false. When the program completes if there is a non-zero value on the stack then it returns true and if a zero value or the stack is empty it will return false. If the stack has more than one value the program also returns false. The following diagram illustrates how the stack machine processes the program.
+The primary purpose of a TEAL program is to return either true or false. When the program completes if there is a non-zero value on the stack then it returns true and if a zero value or the stack is empty it will return false. If the stack has more than one value the program also returns false unless the `return` opcode is used. The following diagram illustrates how the stack machine processes the program.
 
 Program line number 1:
 
 <center>![Transaction Properties](../../../imgs/teal_overview-2.png)</center>
 <center>*Getting Transaction Properties*</center>
 
-The program uses the `txn` to reference the current transaction lists of properties. Grouped transaction properties are referenced using `gtxn`. The number of transactions in a grouped transaction is available in the global variable `Group Size`. To get the first transaction's receiver use `gtxn 0 Receiver`. See [TEAL Specification Reference](../../../reference/teal/specification.md) for more transaction properties.
+The program uses the `txn` to reference the current transaction lists of properties. Grouped transaction properties are referenced using `gtxn` and `gtxns`. The number of transactions in a grouped transaction is available in the global variable `GroupSize`. To get the first transaction's receiver use `gtxn 0 Receiver`. See [TEAL Specification Reference](../../../reference/teal/specification.md) for more transaction properties.
```

Comment:
"The program uses the `txn` to reference the current transaction lists of properties. Grouped transaction properties are referenced using `gtxn` and `gtxns`. The number of transactions in a grouped transaction is available in the global variable `GroupSize`." --> "The program uses the `txn` field to reference the current transaction's list of properties. Grouped transaction properties are referenced using `gtxn` and `gtxns`. The number of transactions in a grouped transaction is available in the global variable `GroupSize`."